### PR TITLE
[test][scalability] Retry performance test on failure

### LINKED ISSUE #2970: [Flaky scalability test] TestScalability/ClusterQueueClasses
/kind flake

**What happened**:

The `TestScalability/ClusterQueueClasses` test failed on the main periodic branch: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-kueue-test-scheduling-perf-main/1830620553732427776

**What you expected to happen**:

No random failures

**How to reproduce it (as minimally and precisely as possible)**:

Repeat the build on CI.

**Anything else we need to know?**:

```
{Failed  === RUN   TestScalability/ClusterQueueClasses/cq
    checker_test.go:103: Usage 54.71% is less then expected 55.00%
--- FAIL: TestScalability/ClusterQueueClasses/cq (0.00s)
}
```

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind flake

#### What this PR does / why we need it:

Retry running the performance scheduler tests on failure to avoid infra related false positive. 

#### Which issue(s) this PR fixes:

Fixes #2970

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
Overall, I don't like the idea of blindly retrying, but it's ok for bringing stability to this suite.

File: hack/performannce-retry.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+for i in $(seq 1 "$1"); do
+	echo "Try ${i}"
+	err=0
+	make test-performance-scheduler-once || err=$?
+	if [ $err -eq 0 ]; then
+		break
+	else
+		if [ "$i" -lt "$1" ]; then
+			mv "${ARTIFACTS}/junit.xml" "${ARTIFACTS}/junit-fail-${i}.xml" || echo "Unable to copy ${ARTIFACTS}/junit.xml"
```
Comment (mimowo) [inline hack/performannce-retry.sh:35]: If the first run fails, do we get artifacts (logs, metrics, etc.) for it to investigate?

**Review by mimowo — COMMENTED** [review]
I'm ok with this change - it should increase the stability of the suite, as we still get occasional failures.

File: hack/performannce-retry.sh (hunk 1 above)
Comment (trasc) [reply hack/performannce-retry.sh:35]: We can back-up them as well if you thinks it will bring any value, but in my opinion the junit file is sufficient.

Comment (trasc) [reply hack/performannce-retry.sh]: done

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve
Overall I like the change. I believe this will solve the issue with random failures. 

The possible downside might be that we miss a small performance regression which would be compensated by the retries. However, I think the significant regressions we care about will still be detectable even after retries.

**Pull request merged**